File: backend/runner/proxy/testdata/kotlin/echo/src/main/kotlin/xyz/block/ftl/java/test/subscriber/Echo.kt
package xyz.block.ftl.java.test.subscriber

import io.quarkus.logging.Log
import xyz.block.ftl.*
import java.util.concurrent.atomic.AtomicInteger


@Verb
@Export
fun echo(message: String, identity: WorkloadIdentity): String {
  return message + " " + identity.spiffeID()
}

File: backend/runner/proxy/testdata/go/echo/echo.go
package echo

import (
	"context"

	"github.com/block/ftl/go-runtime/ftl"
)

//ftl:verb export
func Echo(ctx context.Context, req string) (string, error) {
	identity := ftl.WorkloadIdentity(ctx)
	id, err := identity.SpiffeID()
	if err != nil {
		return "", err
	}
	return req + " " + id.String(), nil
}
